src/bin/pg_autoctl/pgsql.c

```diff
@@ -2505,7 +2506,11 @@ pgsql_get_postgres_metadata(PGSQL *pgsql,
 		" then coalesce(pg_last_wal_receive_lsn(), pg_last_wal_replay_lsn())"
```

Comment:
Why do we use `pg_last_wal_receive_lsn` here? It seems `pg_last_wal_replay_lsn` is the one that reflects what we want the most.

Reply:
If memory serves, that's because that's the LSN that we can reach when restarting Postgres before promotion or for REPORT_LSN. It's safe on-disk even if it's not applied yet. Applying might have delays because of read-only traffic.